dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BundlePrimaryBitstreamLinkRepository.java

```diff
@@ -53,4 +65,92 @@ public BitstreamRest getPrimaryBitstream(@Nullable HttpServletRequest request,
             throw new RuntimeException(e);
         }
     }
+
+    /**
+     * Sets a primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     * @param bitstream     The Bitstream to use as primaryBitstream
+     * @param projection    The projection to use
+     * @return              The Bundle
+     */
+    @PreAuthorize("hasPermission(#bundleId, 'BUNDLE', 'WRITE')")
+    public BundleRest createPrimaryBitstream(Context context, UUID bundleId,
+                                             Bitstream bitstream, Projection projection) {
+        try {
+            Bundle bundle = setPrimaryBitstream(context, bundleId, bitstream, true);
+            return converter.toRest(context.reloadEntity(bundle), projection);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Updates a primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     * @param bitstream     The Bitstream to use as primaryBitstream
+     * @param projection    The projection to use
+     * @return              The Bundle
+     */
+    @PreAuthorize("hasPermission(#bundleId, 'BUNDLE', 'WRITE')")
+    public BundleRest updatePrimaryBitstream(Context context, UUID bundleId,
+                                             Bitstream bitstream, Projection projection) {
+        try {
+            Bundle bundle = setPrimaryBitstream(context, bundleId, bitstream, false);
+            return converter.toRest(context.reloadEntity(bundle), projection);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Deletes the primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     */
+    @PreAuthorize("hasPermission(#bundleId, 'BUNDLE', 'WRITE')")
+    public void deletePrimaryBitstream(Context context, UUID bundleId) {
+        try {
+            Bundle bundle = setPrimaryBitstream(context, bundleId, null, false);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Internal method to set the primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     * @param bitstream     The Bitstream to use as primaryBitstream
+     * @param shouldBeSet   Whether a primaryBitstream is should already be set
```

Comment:
I found the message a bit confusing, what about: assess the current status of the primary bitstream for the bundle. If the expectation is not meet the method will throw a UnprocessableEntityException